app/autodiscovery/thunderbird/src/main/java/com/fsck/k9/autodiscovery/thunderbird/ThunderbirdAutoconfigFetcher.kt

```diff
@@ -8,13 +9,17 @@ import okhttp3.Request
 class ThunderbirdAutoconfigFetcher(private val okHttpClient: OkHttpClient) {
 
     fun fetchAutoconfigFile(url: HttpUrl): InputStream? {
-        val request = Request.Builder().url(url).build()
+        return try {
+            val request = Request.Builder().url(url).build()
 
-        val response = okHttpClient.newCall(request).execute()
+            val response = okHttpClient.newCall(request).execute()
 
-        return if (response.isSuccessful) {
-            response.body?.byteStream()
-        } else {
+            if (response.isSuccessful) {
+                response.body?.byteStream()
+            } else {
+                null
+            }
+        } catch (e: IOException) {
             null
```

Comment:
Detekt flagged that this will swallow the exception. Please add a log statement like the following:

```kotlin
Timber.d(e, "Error fetching URL: %s", url)
```